pennylane_qiskit/qiskit_device2.py

```diff
@@ -98,7 +98,34 @@ def circuit(x):
     """
     # Code to acquire session:
     existing_session = device._session
-    session = Session(backend=device.backend)
+
+    # When an existing session exists, we want to use its settings unless overwritten
+    # by settings in the qiskit_session
+    if existing_session:
+        session_args = inspect.signature(Session).parameters
+        session_options = {arg: getattr(existing_session, "_" + arg) for arg in session_args}
+    else:
+        # when an existing session doesn't exist, we create a session with default settings
+        session_options = {"backend": device.backend, "service": device.service}
+
+    for k, v in kwargs.items():
+        # Options like service and backend should be tied to the settings set on device
+        if k in session_options and k != "max_time":
+            warnings.warn(f"Using '{k}' set in device, {getattr(device, k)}", UserWarning)
+
+        # Need "_" since `max_time` attribute on Session is `_max_time`
+        # When there is overlap between the Session options on the device and the session options
+        # passed in via qiskit_session, we prefer the ones passed in via qiskit_session
+        elif existing_session and getattr(existing_session, "_" + k):
```

Comment:
Do we need to worry about setting a default for the `getattr` here?

Reply:
I don't think so. `existing_session`, if it exists, should always be some `Session()` which has all those attributes. I suppose if the User themselves wrote their own version of `Session` (e.g. our mock) this could be a problem, but I don't see why a User would want to do that...? I suppose if Qiskit significantly changes the attributes of Session or adds an attribute without an "_" in front of it this would be a problem. I guess it wouldn't hurt to set a default :shrug: but not sure how much value it provides.